pkg/apis/batch/validation/validation.go

```diff
@@ -168,6 +170,58 @@ func validateJobSpec(spec *batch.JobSpec, fldPath *field.Path, opts apivalidatio
 		}
 	}
 
+	if spec.PodFailurePolicy != nil {
+		containerNames := sets.NewString()
+		for _, containerSpec := range spec.Template.Spec.Containers {
+			containerNames.Insert(containerSpec.Name)
+		}
+		for _, containerSpec := range spec.Template.Spec.InitContainers {
+			containerNames.Insert(containerSpec.Name)
+		}
+		for i, rule := range spec.PodFailurePolicy.Rules {
+			rulePath := fldPath.Child("podFailurePolicy").Index(i)
+
+			if rule.Action != batch.PodFailurePolicyActionIgnore && rule.Action != batch.PodFailurePolicyActionFailJob && rule.Action != batch.PodFailurePolicyActionCount {
+				actionPath := rulePath.Child("action")
+				allErrs = append(allErrs, field.NotSupported(actionPath, rule.Action, []string{string(batch.PodFailurePolicyActionIgnore), string(batch.PodFailurePolicyActionFailJob), string(batch.PodFailurePolicyActionCount)}))
+			}
```

Comment:
a couple tweaks here:

1. before checking if the value is allowed, make sure they specified a value for rule.Action, and add a field.Required error if they didn't
2. this can be simplified by making a private package-level `sets.String` of the supported values, check `if !supportedPodFailurePolicyActions.Has(string(rule.Action))` and use `field.NotSupported(..., supportedPodFailurePolicyActions.List())` to get ordered []string output for the error

(similar feedback on the Operator field validation below - check present first, use a stringset for checking it is a valid value and error message construction)

Reply:
Updated